beacon-chain/rpc/eth/debug/debug.go

```diff
@@ -26,11 +25,10 @@ func (ds *Server) GetBeaconState(ctx context.Context, req *ethpbv1.StateRequest)
 		return nil, helpers.PrepareStateFetchGRPCError(err)
 	}
 
-	st, ok := beaconSt.(*statev1.BeaconState)
-	if !ok {
-		return nil, status.Error(codes.Internal, "State type assertion failed")
+	if beaconSt.Version() != version.Phase0 {
```

Comment:
This will not work for altair/bellatrix states ?

Reply:
Or correct me if I am wrong, this is our soon to be deprecated rpc method ?

Reply:
We have `GetBeaconStateV2` that works for both phase 0 and Altair (and should work for Bellatrix as well)